### ISSUE TITLE
Difference between the paper and the code

### ISSUE THREAD

**Issue by Haelles** [SHI-Labs/Compact-Transformers#4]
Hi, Steven Walton
Thank you for sharing your amazing work!

If feasible, I would like to ask a question.
In the paper I found that "*ViT, by default, uses a dropout for the MLP heads only with a probability of 0.1, and no attention dropout.Conversely, we do not use MLP dropout and only use attention dropout (p = 0.1).*" However, in utils/transformers.py I found:
````
src_temp = self.activation(self.linear1(src))
src2 = self.linear2(self.dropout1(src_temp))
src = src + self.drop_path(self.dropout2(src2))
````
I wonder whether my understanding is wrong. I’d really appreciate it if you can let me know.

**Comment by stevenwalton** [comment]
Hey Haelles,

If you look at [src/cct:231](https://github.com/SHI-Labs/Compact-Transformers/blob/f6d43e50ece006b933eeb27b087a0c3cad3bc635/src/cct.py#L231) you'll see that we hard code CCT (and CVT) to have a dropout of 0. Though ViTLite [line 161](https://github.com/SHI-Labs/Compact-Transformers/blob/f6d43e50ece006b933eeb27b087a0c3cad3bc635/src/cct.py#L161) has a dropout of 0.1. 

The lines that you are looking at ([src/transformers.py:62](https://github.com/SHI-Labs/Compact-Transformers/blob/main/src/transformers.py#L62)) are called by all the models. They all use the same Transformer Encoder structure. But CCT and CVT will not apply dropout and you can safely ignore that for those models, unless you want to do some hyper parameter testing yourself. 

I can see the ambiguity in how we wrote it. We'll make sure this is more clear in the next version of the paper. Thank you!

**Comment by Haelles** [comment]
> Hey Haelles,
> 
> If you look at [src/cct:231](https://github.com/SHI-Labs/Compact-Transformers/blob/f6d43e50ece006b933eeb27b087a0c3cad3bc635/src/cct.py#L231) you'll see that we hard code CCT (and CVT) to have a dropout of 0. Though ViTLite [line 161](https://github.com/SHI-Labs/Compact-Transformers/blob/f6d43e50ece006b933eeb27b087a0c3cad3bc635/src/cct.py#L161) has a dropout of 0.1.
> 
> The lines that you are looking at ([src/transformers.py:62](https://github.com/SHI-Labs/Compact-Transformers/blob/main/src/transformers.py#L62)) are called by all the models. They all use the same Transformer Encoder structure. But CCT and CVT will not apply dropout and you can safely ignore that for those models, unless you want to do some hyper parameter testing yourself.
> 
> I can see the ambiguity in how we wrote it. We'll make sure this is more clear in the next version of the paper. Thank you!

Thank you for your reply.
If I may ask again, I added `print(self.dropout1)` at [src/transformers.py:62](https://github.com/SHI-Labs/Compact-Transformers/blob/f6d43e50ece006b933eeb27b087a0c3cad3bc635/src/transformers.py#L62) and got `Dropout(p=0.1, inplace=False)`. I think there is something wrong here [src/cct.py:109](https://github.com/SHI-Labs/Compact-Transformers/blob/f6d43e50ece006b933eeb27b087a0c3cad3bc635/src/cct.py#L109), the value of parameter "dropout" should be "dropout" rather than "dropout_rate". The value of "dropout_rate" is 0.1 here.
Really appreciate your help!

**Comment by alihassanijr** [comment]
Hello @Haelles 

Thank you for bringing this to our attention. We noticed there was a small key mismatch in this version, and we just fixed it in the latest commit (619f4bee334c132b444438e893df2cb7180a082d).

**Comment by Haelles** [comment]
Thanks for your response!
